File: app/src/sample/java/com/mycompany/myapp/data/api/github/model/Commit.java
package com.mycompany.myapp.data.api.github.model;

import com.google.gson.annotations.SerializedName;

import org.parceler.Parcel;

@Parcel
public class Commit {
    @Parcel
    public static class Author {
        @SerializedName("name") String name;
    }

    @Parcel
    public static class CommitDetails {
        @SerializedName("message") String message;
        @SerializedName("author") Author author;
    }

    @SerializedName("commit") CommitDetails commit;

    public String getCommitMessage() {
        return commit.message;
    }

    public String getAuthor() {
        return commit.author.name;
    }
}

File: app/src/main/java/com/atomicrobot/carbon/data/api/github/model/Commit.kt
package com.atomicrobot.carbon.data.api.github.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class Author(
    @Json(name = "name") val name: String,
    @Json(name = "email") val email: String,
    @Json(name = "date") val date: String
)

@JsonClass(generateAdapter = true)
data class CommitDetails(
    @Json(name = "message") val message: String,
    @Json(name = "author") val author: Author,
)

@JsonClass(generateAdapter = true)
data class Commit(
    @Json(name = "commit") val commit: CommitDetails,
    @Json(name = "sha") val sha: String
) {

    val commitMessage: String
        get() = commit.message

    val author: String
        get() = commit.author.name
}
